cpp/src/unary/math_ops.cu

```diff
@@ -35,47 +35,47 @@ namespace detail {
 struct DeviceSin {
     template<typename T>
     __device__
-    T apply(T data) {
+    T operator()(T data) {
         return std::sin(data);
```

Comment:
@harrism currently these are using the `std` operators, which works in device code because they are `constexpr`.

What are your thoughts on if we should instead be using the versions of these operators from the CUDA math lib? https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__SINGLE.html

Reply:
I'm guessing that in device code that std::sin resolves to the CUDA math library's sin() anyway? Otherwise how would it work?

Reply:
I think the advantage of the `std` versions is that there are overloads for `float` and `double` so we can get the most efficient code without explicitly calling `sinf`, for example.

Reply:
Yeah, looks like they're identical: https://godbolt.org/z/T4fN_i